src/transformers/pipelines/visual_question_answering.py

```diff
@@ -33,17 +73,28 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.check_model_type(MODEL_FOR_VISUAL_QUESTION_ANSWERING_MAPPING)
 
-    def _sanitize_parameters(self, top_k=None, padding=None, truncation=None, **kwargs):
+    def _sanitize_parameters(self, top_k=None, padding=None, truncation=None, words=None, boxes=None, **kwargs):
         preprocess_params, postprocess_params = {}, {}
         if padding is not None:
             preprocess_params["padding"] = padding
         if truncation is not None:
             preprocess_params["truncation"] = truncation
         if top_k is not None:
             postprocess_params["top_k"] = top_k
+        if words is not None or boxes is not None:
+            if words is None or boxes is None:
+                raise ValueError("Must provide both words and boxes if providing either")
+            preprocess_params["words"] = words
+            preprocess_params["boxes"] = boxes
+
         return preprocess_params, {}, postprocess_params
 
-    def __call__(self, image: Union["Image.Image", str], question: str = None, **kwargs):
+    def __call__(
+        self,
+        image: [Optional[Union["Image.Image", str]]] = None,
+        question: Optional[Union[List[str], str]] = None,
```

Comment:
This `None` here exists solely to support `pipeline({"image": image, "question": question})` (Which is necessary when you want to work with datasets or lists directly).

It's not documented kind of on purpose, the documentation bodies showcases the various uses, but here the strict-typing just causes things to be less readable IMO, since if you do send it it should be a `str`.

Reply:
I read the documentation below and the `List` options for `question` and `image` are actually wrong and shouldn't be there.

Reply:
Okay cool. I'll just revert this since we'll place things in a new pipeline anyway.

Reply:
Feel free to do the typing in the new pipeline. The most critical piece is the docstring (and this is the one where readability > completeness)